indigo_app/views/tasks.py

```diff
@@ -164,13 +168,33 @@ class TaskEditView(TaskViewBase, UpdateView):
 
     def form_valid(self, form):
         task = self.object
+        old_workflows = [wf.id for wf in task.workflows.all()]
         task.updated_by_user = self.request.user
-        action.send(self.request.user, verb='updated', action_object=task,
-                    place_code=task.place.place_code)
         task.workflows = form.cleaned_data.get('workflows')
 
-        # TODO: send action signal saying 'User added|removed Task to|from Workflow,
-        #  and don't send  generic 'updated' signal if this was the only change
+        # action signals
+        # first, was something changed other than workflows?
+        if form.changed_data:
+            action.send(self.request.user, verb='updated', action_object=task,
+                        place_code=task.place.place_code)
+        # then, was the task added to / removed from any workflows?
+        new_workflows = [wf.id for wf in task.workflows.all()]
+        if set(old_workflows) != set(new_workflows):
```

Comment:
You can also do set operations: `set(old_workflows) - set(new_workflows)` is all the ones that have been removed. `new - old` are the ones that are new. `old & new` is the intersection, those that haven't changed.

See https://docs.python.org/2/library/sets.html

Reply:
this is amazing thank you!